code/HabitTracker/app/src/main/java/com/example/habittracker/HabitActivity.java

```diff
@@ -13,8 +25,72 @@
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_habit);
+
+        // TEST DATABASE
+        testDatabase();
+      
         NavBarManager nav = new NavBarManager(this,findViewById(R.id.bottom_navigation));
         Intent intent = new Intent(this, HomeActivity.class);
         startActivity(intent);
     }
+
+    void testDatabase() {
+        testCreateUser1();
+        testUpdateUser1();
+//        testDeleteUser1();
+    }
+
+    // demonstrates how to create a document with the DatabaseManager class
+    void testCreateUser1() {
+        HashMap<String, Object> userDocument = new HashMap<>();
+        HashMap<String, Object> habitDocument = new HashMap<>();
+        HashMap<String, Object> habitEventDocument = new HashMap<>();
+
+        // add data for the user document
+        String userid = "user1";
+        userDocument.put("following", Arrays.asList("user2"));
+        userDocument.put("followers", Arrays.asList("user2"));
+        userDocument.put("pendingFollowReqs", Arrays.asList("user3"));
+        userDocument.put("pendingFollowerReqs", Arrays.asList("user3"));
+
+        // use database manager
+        DatabaseManager.get().addUsersDocument(userid, userDocument);
+
+        // add data for the habit document
+        String title = "Habit 1";
+        habitDocument.put("reason", "Reason 1");
+        habitDocument.put("dateStarted", Arrays.asList(2021, 10, 22));  // YYYY/MM/DD
+        habitDocument.put("whatDays", Arrays.asList("Monday", "Wednesday", "Saturday"));
+        habitDocument.put("progress", 0);
+
+        // use database manager
+        DatabaseManager.get().addHabitDocument(userid, title, habitDocument);
+
+        // add data for the habit event document
+        habitEventDocument.put("comment", "Comment 1");
+        habitEventDocument.put("image", "Image 1");
+        habitEventDocument.put("location", "Location 1");
+
+        // use database manager
+        DatabaseManager.get().addHabitEventDocument(userid, title, habitEventDocument);
+    }
+
+    // demonstrates how to update particular documents in Firestore
+    void testUpdateUser1() {
+        // add a new user to the following list
+        DatabaseManager.get().getUsersColRef().document("user1")
+                .update("following", FieldValue.arrayUnion("user7"))
+                .addOnSuccessListener(new OnSuccessListener<Void>() {
+                    @Override
+                    public void onSuccess(Void unused) {
+                        Log.d("HabitActivity", "Following list updated");
+                    }
+                });
+    }
+
+    // demonstrates how to delete a user with the DatabaseManager class
+    void testDeleteUser1() {
```

Comment:
Something to note. And maybe something to discuss for later. (We might wanna make an issue out of this, maybe, since deleting can get quite complicated)

Deleting a user is going to be more than just deleting the user document. You will probably have to search and delete all other references made to the deleted user.

For example:
When a user X is deleted, and X had been in the followers list of User Y. What happens? I believe, since X does not exist anymore, they should no longer appear in the follower list of any other users. Same for request list and all other lists, really.

Reply:
Yes. I was thinking of doing that work in future by making an issue out of it, since it gets kinda complicated.